warehouse/query-core/src/main/java/datawave/query/config/RemoteQueryConfiguration.java

```diff
@@ -0,0 +1,153 @@
+     * @return - a clean RemoteQueryConfiguration
+     */
+    public static RemoteQueryConfiguration create() {
+        return new RemoteQueryConfiguration();
+    }
+    
+    /**
+     * Factory method that returns a deep copy of the provided RemoteQueryConfiguration
+     *
+     * @param other
+     *            - another instance of a RemoteQueryConfiguration
+     * @return - copy of provided RemoteQueryConfiguration
+     */
+    public static RemoteQueryConfiguration create(RemoteQueryConfiguration other) {
+        return new RemoteQueryConfiguration(other);
+    }
+    
+    /**
+     * Factory method that creates a RemoteQueryConfiguration deep copy from a RemoteQueryLogic
+     *
+     * @param remoteQueryLogic
+     *            - a configured RemoteQueryLogic
+     * @return - a RemoteQueryConfiguration
+     */
+    public static RemoteQueryConfiguration create(RemoteEventQueryLogic remoteQueryLogic) {
+        return create(remoteQueryLogic.getConfig());
+    }
+    
+    /**
+     * Factory method that creates a RemoteQueryConfiguration from a RemoteQueryLogic and a Query
+     *
+     * @param remoteQueryLogic
+     *            - a configured RemoteQueryLogic
+     * @param query
+     *            - a configured Query object
+     * @return - a RemoteQueryConfiguration
+     */
+    public static RemoteQueryConfiguration create(RemoteEventQueryLogic remoteQueryLogic, Query query) {
+        RemoteQueryConfiguration config = create(remoteQueryLogic);
+        config.setQuery(query);
+        return config;
+    }
+    
+    public String getRemoteId() {
+        return remoteId;
+    }
+    
+    public void setRemoteId(String remoteId) {
+        this.remoteId = remoteId;
+    }
+    
+    public String getRemoteQueryLogic() {
+        return remoteQueryLogic;
+    }
+    
+    public void setRemoteQueryLogic(String remoteQueryLogic) {
+        this.remoteQueryLogic = remoteQueryLogic;
+    }
+    
+    public Query getQuery() {
+        return query;
+    }
+    
+    public void setQuery(Query query) {
+        this.query = query;
+    }
+    
+    @Override
+    public boolean equals(Object o) {
+        if (this == o)
+            return true;
+        if (o == null || getClass() != o.getClass())
+            return false;
+        if (!super.equals(o))
+            return false;
+        RemoteQueryConfiguration that = (RemoteQueryConfiguration) o;
+        return Objects.equals(getRemoteId(), that.getRemoteId()) && Objects.equals(getRemoteQueryLogic(), that.getRemoteQueryLogic())
+                        && Objects.equals(getQuery(), that.getQuery());
+    }
+    
+    @Override
+    public int hashCode() {
+        return Objects.hash(super.hashCode(), getRemoteId(), getRemoteQueryLogic(), getQuery());
+    }
+    
+    // Part of the Serializable interface used to initialize any transient members during deserialization
+    protected Object readResolve() throws ObjectStreamException {
+        return this;
+    }
```

Comment:
Is this strictly necessary since there are no transient members _or_ should this have the `@Override` annotation because you are overriding behavior from a superclass?